### DISCUSSION TITLE
Question: Why don't fugitive buffers have `acwrite` `'buftype'`?

### DISCUSSION THREAD

**Discussion by tmillr** [tpope/vim-fugitive#2159] [Q&A] [answered]
Usually you can easily detect whether a buffer will use autoccmds for writing just by checking its `'buftype'`; this doesn't seem to work with fugitive however, as it doesn't set `'buftype'` to `acwrite` for its `BufWriteCmd` buffers. Why not?

**Comment by tpope** [comment, marked as answer]
All `buftype=acwrite` does is tell Vim that the buffer name is "unrelated to a file". That is, it's identical to `buftype=nofile`, except it doesn't block `:write` from working. This means, for example, that the buffer name won't be written to a session file, because if it's not related to a file, then you wouldn't be able to `:edit` it. But you _can_ edit `fugitive://` file names. As far as Vim is concerned, they _are_ related to a (virtual) file.

An alternative you might find helpful (depending on your use case) is to look for buffer names that look like URLs. Try `@% =~# '^\a\+:[\/][\/]'` (the backslashes are necessary to match on Windows).

**Reply by tmillr** [reply]
Thank you! I learned something new. I wish more authors were this friendly and helpful, and cared this much about their users. Definitely worth a donation.
